dask_image/ndinterp/__init__.py

```diff
@@ -0,0 +1,171 @@
+        - modes 'reflect', 'mirror' and 'wrap' may yield different results
+
+        To do:
+        - optionally use cupyx.scipy.ndimage.affine_transform
+
+        Arguments equal to `ndimage.affine_transformation`,
+        except for `output_chunks`.
+
+    Parameters
+    ----------
+    input : array_like (Numpy Array, Dask Array, ...)
+        The input array.
+    matrix : array (ndim,) or (ndim, ndim)
+        Transformation matrix.
+    offset : array (ndim,)
+        Transformation offset.
+    output_shape : array (ndim,), optional
+        The size of the array to be returned.
+    output_chunks : array (ndim,), optional
+        The chunks of the output Dask Array.
+
+    Returns
+    -------
+    affine_transform : Dask Array
+        A dask array representing the transformed output
+
+    """
+
+    if output_shape is None:
+        output_shape = input.shape
+
+    # process kwargs
+
+    # set default order to 1, warn if different
+    if 'order' in kwargs:
+        order = kwargs['order']
+        del kwargs['order']
+        if order > 1:
+            UserWarning('Currently, for order > 1 the output of'
+                        '`dask_image.ndinterp.affine_transform` can differ'
+                        'from that of `ndimage.affine_transform`')
+    else:
+        order = 1
+
+    # prefilter is not yet supported
+    if 'prefilter' in kwargs:
+        if kwargs['prefilter'] and order > 1:
+            UserWarning('Currently, `dask_image.ndinterp.affine_transform`'
+                        'does not support `prefilter=True`')
+        del kwargs['prefilter']
+
+    transformed = da.zeros(output_shape,
+                           dtype=input.dtype,
+                           chunks=output_chunks)
+
+    transformed = transformed.map_blocks(resample_chunk,
+                                         dtype=input.dtype,
+                                         input=input,
+                                         matrix=matrix,
+                                         offset=offset,
+                                         order=order,
+                                         func_kwargs=kwargs,
+                                         )
+
+    return transformed
+
+
+def resample_chunk(chunk, input, matrix, offset, order, func_kwargs, block_info=None):
+    """Resample a given chunk, used by `affine_transform`."""
+
+    n = chunk.ndim
+    input_shape = input.shape
+    chunk_shape = chunk.shape
+
+    chunk_offset = [i[0] for i in block_info[0]['array-location']]
+
+    chunk_edges = np.array([i for i in np.ndindex(tuple([2] * n))])\
+        * np.array(chunk_shape) + np.array(chunk_offset)
+
+    rel_input_edges = np.dot(matrix, chunk_edges.T).T + offset
+    rel_input_i = np.min(rel_input_edges, 0)
+    rel_input_f = np.max(rel_input_edges, 0)
+
+    # expand relevant input to contain footprint of the spline kernel
+    # according to
+    #
+    # https://github.com/scipy/scipy/blob/
```

Comment:
I'm happy for comment lines with long urls on them to extend past 80 characters, and would prefer it since it's hard to copy-paste a url that runs over multiple lines. I think you'd just need to tell flake8 to ignore it by adding a ` # noqa: E501` comment to the end of the line.

Reply:
That makes sense, didn't know this trick :)